molvis/molvisbase/src/util/molvisutils.cpp

```diff
@@ -0,0 +1,105 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/molvisbase/util/molvisutils.h>
+
+#include <inviwo/core/datastructures/geometry/mesh.h>
+#include <inviwo/core/datastructures/buffer/bufferram.h>
+
+#include <inviwo/core/util/stdextensions.h>
+
+namespace inviwo {
+
+namespace molvis {
+
+namespace {
+
+template <typename T>
+T square(T a) {
+    return a * a;
+}
+
+}  // namespace
+
+bool covalentBondHeuristics(Element element1, const dvec3& pos1, Element element2,
+                            const dvec3& pos2) {
+    const dvec3 delta{pos2 - pos1};
+    const double distSq = glm::dot(delta, delta);
+
+    const double covalentDist =
+        element::covalentRadius(element1) + element::covalentRadius(element2);
+    const double dMinSq = square(covalentDist - 0.5);
+    const double dMaxSq = square(covalentDist + 0.3);
+    return (dMinSq < distSq) && (distSq < dMaxSq);
+}
+
+std::shared_ptr<Mesh> createMesh(const MolecularStructure& s) {
+    std::vector<vec3> positions{
+        util::transform(s.getAtoms().positions, [](auto& p) { return vec3{p}; })};
+    std::vector<vec4> colors;
+    std::vector<float> radius;
+    // std::vector<uint32_t> picking;
```

Comment:
remove or add picking as optional?

Reply:
remove picking for now as this would require a PickingManager. Might add at a later point.